### PR TITLE
remove signature in details and rename subfulfillments to subconditions

### PR DESCRIPTION
**Pull request by libscott**
Update to go with https://github.com/bigchaindb/bigchaindb/pull/1589

### CODE REVIEW

File: src/transaction/utils/ccJsonLoad.js [hunk 1]
```diff
@@ -22,22 +22,19 @@ export default function ccJsonLoad(conditionJson) {
...
+                } else if ('serializeUri' in subcondition) {
+                    fulfillment.addSubcondition(subcondition)
                 }
             })
         }
 
         if (conditionJson.type === 'ed25519-sha-256') {
             fulfillment = new cc.Ed25519Sha256()
             fulfillment.publicKey = new Buffer(base58.decode(conditionJson.public_key))
-            if (conditionJson.signature) {
```
Comment (diminator) [inline src/transaction/utils/ccJsonLoad.js]: why is this removed?

Comment (libscott) [inline src/transaction/utils/ccJsonLoad.js]: Well, the `output.condition` doesn't seem like the right place to have a signature.

**Comment by TimDaub** [comment]
> In general I think we should first have our internal marshalling spec before we start changing stuff

@diminator I think your feedback belongs here: https://github.com/bigchaindb/bigchaindb/pull/1589
This PR is just a regression.

**Comment by libscott** [comment]
> In general I think we should first have our internal marshalling spec before we start changing stuff
> 
> it could be that you there actually is a mix of subfulfillments and subconditions that the output wants to publish
> we should make sure there is a way of marshalling topology and attributes
> I havent had too many use cases where I depend on this marshall, as it is assumed in the interledger space that contract negotiation is done outside of the ledger

Why would you need to have a fulfillment inside `output.details`?

In general, re "conditions" vs "fulfillments", the main use case is to specify conditions. If there were a requirement to specify a fulfillment then I'd frame it as a condition that has been fulfilled, rather than a separate type.

If the use case is multi party signing, I still think the output isn't the right place; the transaction schema allows for the input to contain a fulfillment that is a `condition.details` object. The server won't accept this as a fulfillment, but it may be useful for transactions that are only partially fulfilled.

**Comment by libscott** [comment]
> If you change the docker file in .travis.yml to bigchaindb:master, tests might also pass.

Giving it a try; would we want to merge that though?

**Comment by diminator** [comment]
> If the use case is multi party signing, I still think the output isn't the right place; the transaction schema allows  for the input to contain a fulfillment that is a condition.details object. The server won't accept this as a fulfillment, but it may be useful for clients to exchange transactions that are only partially fulfilled.

agreed, since there is a PR in place, I'll let this pass

**Comment by libscott** [comment]
Closed in favour of #72
